tests/unit/util/test_getter.py

```diff
@@ -744,3 +751,56 @@ def test_get_raises_401_if_no_token_is_received(self):
             with mock.patch.dict("os.environ", mock_env):
                 http_getter = GetterFactory.from_string("https://some.url/configuration")
                 http_getter.get()
+
+    @pytest.mark.parametrize(
+        "testcase, credential_file_content, instance",
+        [
+            (
+                "Return BasicAuthCredential object",
+                """---
+"https://some.url":
+    username: test
+    password: test
+""",
+                BasicAuthCredentials,
+            ),
+            (
+                "Return OAuthPasswordFlowCredential object",
+                """---
+"https://some.url":
+    endpoint: https://endpoint.end
+    username: test
+    password: test
+""",
+                OAuth2PasswordFlowCredentials,
+            ),
+            (
+                "Return OAuthClientFlowCredential object",
+                """---
+"https://some.url":
+    endpoint: https://endpoint.end
+    client_id: test
+    client_secret: test
+""",
+                OAuth2ClientFlowCredentials,
+            ),
+            (
+                "Return OAuthTokenCredential object",
+                """---
+"https://some.url":
+    token: "jsoskdmoiewjdoeijkxsmoiqw8jdiowd0"
+""",
+                OAuth2TokenCredentials,
+            ),
+        ],
+    )
+    def test_credentials_returns_credential_object(
```

Comment:
we need tests for maleformed credential files. and tests for json based credential files